src/openassetio-python/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+#-----------------------------------------------------------------------
+# Python module target
+
+add_library(openassetio-python MODULE)
+set_default_target_properties(openassetio-python)
+
+# Add to the set of installable targets.
+install(
+    TARGETS openassetio-python
+    EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS
+    DESTINATION ${OPENASSETIO_PYTHON_SITEDIR}
+)
+
+# Set Python extension module name, and remove any ("lib") prefix.
+set_target_properties(openassetio-python
+    PROPERTIES
+    OUTPUT_NAME _openassetio
+    PREFIX "")
+
+
+#-----------------------------------------------------------------------
+# Target dependencies
+
+target_sources(openassetio-python PRIVATE _openassetio.cpp)
+
+target_link_libraries(openassetio-python
+    PRIVATE
+    # Core C++ library.
+    openassetio-core
+    # pybind, including its handy transitive Python-specific properties.
+    pybind11::module pybind11::windows_extras
+)
+
+#-----------------------------------------------------------------------
+# Override build tree to look like install tree.
+
+set_target_properties(openassetio-python
+    PROPERTIES
+    OUTPUT_NAME _openassetio
+    RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${OPENASSETIO_PYTHON_SITEDIR}
+    LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${OPENASSETIO_PYTHON_SITEDIR})
+
+
+#-----------------------------------------------------------------------
+# Override RPATH in (usual) case that Python .so and core .so live in
+# different locations.
+
+# Calculate relative path from site-packages to lib directory.
+file(RELATIVE_PATH
+    python_site_dir_rel_to_lib
+    "${CMAKE_INSTALL_PREFIX}/${OPENASSETIO_PYTHON_SITEDIR}"
+    "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+
+set_target_properties(
+    openassetio-python
+    PROPERTIES
+    INSTALL_RPATH "$ORIGIN/${python_site_dir_rel_to_lib}"
```

Comment:
[consider] This might only work on Linux, probably is worth to branch out doing the right thing per OS in here (gotta love the rpath fun! :smile: )

Reply:
Good point, added OS-specific branching and TODO comment in f1ed0ce9